frontend/src/components/Login.js

```diff
@@ -45,42 +53,52 @@ function Login() {
     }
   };
 
-  function makeTextInput (id,name,placeholder){
-    return(<input
-              className="form-control"
-              type="text"
-              id={id}
-              name={name}
-              placeholder={placeholder}
-            />
-          )
+  function makeInput(type, id, name, placeholder) {
+    return (
+      <div className="has-validation">
+        <input
+          className="form-control"
+          type={type}
+          id={id}
+          name={name}
+          placeholder={placeholder}
+        />
+        <div className="invalid-feedback">{name} cannot be blank</div>
+      </div>
+    );
```

Comment:
Here I think you can move the has-validation within the input className and put something descriptive like "verifiedInput" as the outer className